### PR TITLE
added examples and more wordings to documentation of cross entropy loss...

### PR DESCRIPTION
**Pull request by mithunpaul08**
as suggested by developer Richard Zou in issue #[5554](https://github.com/pytorch/pytorch/issues/5554#issuecomment-389630705)

### CODE REVIEW

**Review by zou3519 — CHANGES_REQUESTED** [review]
see comments

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -151,9 +151,15 @@ class NLLLoss(_WeightedLoss):
         - Input: :math:`(N, C)` where `C = number of classes`, or
             :math:`(N, C, d_1, d_2, ..., d_K)` with :math:`K \geq 2`
             in the case of `K`-dimensional loss.
+            Eg:  [0,0,1]
```
Comment (zou3519) [inline torch/nn/modules/loss.py]: The inputs don't have to be one-hot coded. If you want to show an example of the inputs, I think it would be better to put this into the examples section below.

File: torch/nn/modules/loss.py [hunk 2]
```diff
@@ -151,9 +151,15 @@ class NLLLoss(_WeightedLoss):
...
+
         - Target: :math:`(N)` where each value is :math:`0 \leq \text{targets}[i] \leq C-1`, or
             :math:`(N, d_1, d_2, ..., d_K)` with :math:`K \geq 2` in the case of
             K-dimensional loss.
+            Eg:[2]
```
Comment (zou3519) [inline torch/nn/modules/loss.py]: This should probably be a warning somewhere. You can use the `warning` directive:
```
.. warning::
    The target value should be a value indicating what class it belongs to. This is semantically different from the input, which contains log-probabilities for each class.
```

**Comment by zou3519** [comment]
@mithunpaul08 my apologies for missing this! Thank you for the contribution -- I've left some comments above that I think should clarify the docs further :)

**Comment by yf225** [comment]
@mithunpaul08 Let us know if you have time to address @zou3519 's comments and also rebase this PR onto current master. Thanks!

**Comment by mithunpaul08** [comment]
well, a little tight on time since I have a conference deadline looming large. Let me see if I can squeeze some time today. If it doesnt happen by EOD today, it means, I didnt find time and might need help :-( Sorry.

**Comment by zou3519** [comment]
No worries, we can take this PR over
